src/main/java/solutions/siren/join/action/terms/collector/IntegerTermsSet.java

```diff
@@ -162,4 +163,58 @@ private void readFromBytes(BytesRef bytesRef) {
     return TermsByQueryRequest.TermsEncoding.INTEGER;
   }
 
+  @Override
+  public void release() {
+    if (set != null) {
+      set.release();
+    }
+  }
+
+  /**
+   * A {@link IntHashSet} integrated with the {@link CircuitBreaker}. It will adjust the circuit breaker
+   * for every new call to {@link #allocateBuffers(int)}.
+   * <p>
+   * This set must not be reused after a call to {@link #release()}.
```

Comment:
this is internal code, and these classes are not meant to be reused elsewhere at the moment. If people use them for something else, it is a their own risk.